### PR TITLE
feat: return pipeline updated after add/update/delete pipeline permissions

### PR DESCRIPTION
**Pull request by sguiheux**
…

### CODE REVIEW

File: engine/api/project_group.go [hunk 1]
```diff
@@ -306,147 +302,141 @@ func addGroupInProject(w http.ResponseWriter, r *http.Request, db *sql.DB, c *co
...
-				return
-			}
-
-			for _, pip := range pipelines {
-				if permission.AccessToPipeline(sdk.DefaultEnv.ID, pip.ID, c.User, permission.PermissionReadWriteExecute) {
-					inPip, err := group.CheckGroupInPipeline(tx, pip.ID, g.ID)
-					if err != nil {
-						log.Warning("AddGroupInProject: Cannot check if group %s is already in the pipeline %s: %s\n", g.Name, pip.Name, err)
+				} else {
+					if err := group.InsertGroupInApplication(tx, app.ID, g.ID, groupProject.Permission); err != nil {
```
Comment (yesnault) [inline engine/api/project_group.go]: else if

File: engine/api/project_group.go [hunk 2]
```diff
@@ -306,147 +302,141 @@ func addGroupInProject(w http.ResponseWriter, r *http.Request, db *sql.DB, c *co
...
-						}
-					} else {
-						err = group.InsertGroupInEnvironment(tx, env.ID, g.ID, groupProject.Permission)
-						if err != nil {
-							log.Warning("AddGroupInProject: Cannot insert group %s on environment %s: %s\n", g.Name, env.Name, err)
-							WriteError(w, r, err)
-							return
-						}
+				} else {
+					if err := group.InsertGroupInPipeline(tx, pip.ID, g.ID, groupProject.Permission); err != nil {
```
Comment (yesnault) [inline engine/api/project_group.go]: else if

File: engine/api/project_group.go [hunk 3]
```diff
@@ -306,147 +302,141 @@ func addGroupInProject(w http.ResponseWriter, r *http.Request, db *sql.DB, c *co
...
+					return
+				}
+				if inEnv {
+					if err := group.UpdateGroupRoleInEnvironment(tx, p.Key, env.Name, g.Name, groupProject.Permission); err != nil {
+						log.Warning("AddGroupInProject: Cannot update group %s on environment %s: %s\n", g.Name, env.Name, err)
+						WriteError(w, r, err)
+						return
+					}
+				} else {
+					if err := group.InsertGroupInEnvironment(tx, env.ID, g.ID, groupProject.Permission); err != nil {
```
Comment (yesnault) [inline engine/api/project_group.go]: idem

File: engine/api/pipeline_group.go [hunk 4]
```diff
@@ -62,54 +61,61 @@ func updateGroupRoleOnPipelineHandler(w http.ResponseWriter, r *http.Request, db
...
 
+	if err := pipeline.UpdatePipelineLastModified(tx, p); err != nil {
+		log.Warning("updateGroupRoleOnPipelineHandler: Cannot update pipeline last_modified date: %s\n", err)
+		WriteError(w, r, err)
+		return
 	}
-	w.WriteHeader(http.StatusOK)
+
+	err = tx.Commit()
+	if err != nil {
```
Comment (fsamin) [inline engine/api/pipeline_group.go]: one line

**Pull request merged**